core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -671,6 +671,20 @@ private boolean isExistingRelation(Session session, QualifiedObjectName name)
                     }
                     tableColumns.put(entry.getKey(), columns.build());
                 }
+
+                // if view and materialized view names overlap, the materialized view wins
```

Comment:
I think winning here just means overwriting any existing value in tableColumns map.
Since the priority here is MV > View > Table, we're consistent with StatementAnalyzer.Visitor#visitTable ?